lib/matplotlib/backends/backend_webagg_core.py

```diff
@@ -150,17 +144,11 @@ def show(self):
         show()
 
     def draw(self):
-        renderer = self.get_renderer(cleared=True)
-
         self._png_is_old = True
-
-        backend_agg.RendererAgg.lock.acquire()
         try:
-            self.figure.draw(renderer)
+            super().draw()
```

Comment:
... this is pretty different.  Does it do the same thing?  Why just call super draw instead of the figure?  Just trying to understand, not saying its wrong....

Reply:
If you're using Agg, this resolves to https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/backends/backend_agg.py#L414 which gets a cleared renderer, handles the locking and calls `self.figure.draw(self.renderer)`.

Of course the not-so-secret objective here is to make it possible to swap in mplcairo's renderer.  For that, the idea is as usual to push as much the renderer-specific code back to the super-class.

Reply:
Makes sense.  I did find `super().draw()` a bit opaque.; ie. why not just `self.draw()`?

Reply:
well that would be an infinite loop...
super().draw() means (approximately) "call the draw method but don't look it up on this class, look it up on the base class [in this case, FigureCanvasAgg] instead" (check up e.g. "python super mro" for details)

Reply:
ooops.  duh.